### DISCUSSION TITLE
i need a suggestion related to blockchain development.

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#588] [Q&A] [answered]
hey there 
I want to ask that is is compulsory to know about node.js if I am doing blockchain development in foundry?

**Comment by 0xusmanf** [comment]
No, you won't need Node.js knowledge to develop smart contracts in foundry. All you need is solidity and some persistence & determination.

**Comment by khushbusandhu** [comment]
hey @usmanfarooq91 !
Does that work the same as with Node.js  or that would be enough.?

**Reply by 0xusmanf** [reply]
Hi, I don't understand, can you please add more context to your question?

**Reply by khushbusandhu** [reply]
Hey @usmanfarooq91 !
i want to ask that which is better like using Node.js or Foundry for development?

**Reply by 0xusmanf** [reply]
As far as I know, Node.js is an open-source server environment and can be used to deploy or interact with smart contracts and on the other hand Foundry is a smart contract development framework. So there shouldn't be any comparison between them and you should definitely use Foundry for development.

**Comment by Tanmay53** [comment]
With Node.js you will first need to know about **JavaScript**(a programming language) then you would want to learn about testing frameworks like **Hardhat** or **Truffle**.

While **Foundry** is itself a testing framework which uses **Solidity** as the programming language.

**Reply by khushbusandhu** [reply]
Does it mean if I don't use Node.js or javascript, I can also be a smart contract developer by using Foundry.?

**Reply by 0xusmanf** [reply]
It depends on what you are trying to achieve, take a look at what @theirrationalone has mentioned below then it will be much clearer to you. And keep asking if it still doesn't. Meanwhile, as you get more and more info, try to reconstruct your question according to the new info you are gaining.

**Reply by itznishant** [reply]
lol I was just thinking of this but I think no major projects use **truffle** now, isn't it.  I used truffle/web3.js a lot in 21/22.

**Reply by theirrationalone** [reply]
Should not take care about tools 🛠️ & contextual contracts implementation.... 

What matters is, Are you getting the Concepts, 

**Blockchain, Nodes, validators, miners, Gas, censorship resistance, different hawks(😉), NFTs, DEFI, DAO, auditing focused mindset, transactions, signing, Minting, ownership, solidity(in context) etc....**

Tools are getting sophisticated time by time.... You can contribute on them tooo.

Foundry, Hardhat, on abstractions are rocking 🎸🎶🎶

So Please Guys help People Guide People don't Argue!

**Comment by theirrationalone** [comment]
Take a look at this.... [Click here](https://github.com/Cyfrin/foundry-full-course-f23/discussions/522#discussioncomment-6620940)

**Comment by itznishant** [comment, marked as answer]
> hey there I want to ask that is is compulsory to know about node.js if I am doing blockchain development in foundry?

Foundry is very **Solidity** heavy and quite a lot to take for beginners IMO. 
As a non-full-stack developer who worked extensively in Data Science before, I suggest to start with Hardhat based web3 development as it uses a familiar javascript based framework and along with ethers.js can be easier to learn.
Also much easier to integrate with frontend, say a dapp with a react frontend.

**Reply by khushbusandhu** [reply]
hey @itznishant !
Currently, I am working on the @PatrickAlphaC course of solidity using the foundry framework but in future, I will surely take a look at Hardhat based web3 development. BTW thanks for your suggestion .

**Reply by shikhar229169** [reply]
There is one thing with the hardhat course, some of the methods that were used in the course for contract interactions are now removed in the newer versions of packages, so you will face issues while coding along. 
So, just remember to always refer to the docs.

**Comment by itznishant** [comment]
> hey @itznishant ! Currently, I am working on the @PatrickAlphaC course of solidity using the foundry framework but in future, I will surely take a look at Hardhat based web3 development. BTW thanks for your suggestion .

Okay, good luck & continue on with Foundry then, and mark my reply as the answer if we're good.

Patrick also has a Hardhat + Js course, it was mentioned couple times in some Foundry lessons , do check.
